src/core/unplugin.ts

```diff
@@ -29,7 +30,7 @@ export default createUnplugin<Options>((options) => {
         const exclude = config.optimizeDeps?.exclude || []
 
         const imports = new Set((await ctx.unimport.getImports()).map(i => i.from)
-          .filter(i => i.match(/^[a-z@]/) && !exclude.includes(i)))
+          .filter(i => isPackageExists(i) && !exclude.includes(i)))
```

Comment:
```suggestion
          .filter(i => i.match(/^[a-z@]/) && !exclude.includes(i) && isPackageExists(i)))
```

It would still be better and more performant to do early bailout